### PR TITLE
Rewrite docs for physics damping project settings

### LINKED ISSUE #8163: Damping for 2D and 3D is described incorrectly.
**Your Godot version:**
4.1.2.stable but also 4.2.dev.custombuild

**Issue description:**
The documentation is grossly(!) incorrect in describing the linear and angular damping techniques used. It insinuates a **critical spring damper function** (sometimes known as smoothdamp), in particular in the language where it says:

> Note: Good values are in the range 0 to 1. At value 0 objects will keep moving with the same velocity. Values greater than 1 will aim to reduce the velocity to 0 in less than a second e.g. a value of 2 will aim to reduce the velocity to 0 in half a second. 

This is wrong because:
- it's actually implemented as a **negative exponential damper**, which behaves very differently (and poorly, because this implementation is fully tickrate dependent), this warrants special care to be taken to educate the user of pitfalls
- a value of 1 reduces the magnitude to 0 in **more than** the described time of 1 second, never less. In fact, it will barely have removed 65% of the speed at that time, and the rate of deceleration is shrinking)
- it's not trying "to stop in D seconds", that may have been the intent of the programmer but it's not what the function does: especially at the 0.1 default value; which will very definitely not have stopped after anywhere near 10 seconds (the sleep threshold of 0.1m/s is crossed after 23 seconds for a low starting speed of 1m/s, and 46 seconds for a speed fo 10m/s, and a whole minute at 40m/s [90 mph / 145 kph])
- 1 is not a good upper bound for this value, it can easily go as high as 10 or 20 and still make perfect sense. In no way is a value betweeen 0 and 1 "good" with this damper, in fact, it's too low for most intentes and leads to floaty physics
- technically, the 2D physics engine supports values as low as -1 and only below this value will start throwing errors.
- the tickrate dependence means that a value of d=65 will stop an object dead immediately (after a distance of 0) at a tickrate of 60 tps, but at 200 tps, it will keep moving for a timespan of a good 20 frames (speed=1) or much more (speed>1, e.g. 10), and after a definitely nonzero distance that grows significantly with higher speed. (too tired to solve the integral right now)

The source code applies the following operation:
```cpp
real_t damp = 1.0 - p_step * total_linear_damp;
//...
linear_velocity *= damp;
```

This is, after `n` physics ticks, equivalent of 
`linear_velocity = linear_velocity_start * pow(1-p_step*total_linear_damp, n)`

For a value of 2 at a physics simulation frame rate of 60, this will ACTUALLY result in about 18% remaining velocity after 0.5 seconds, and depending on the starting velocity.

At different physics tick rates, this will also result in different behaviour altogether of the damping function; this would be an issue to fix in the engine repo, however; the correct way to do this would be to peg this "damp" value (I usually call it k) to a reference tick rate; calculating k = (1.0*total_linear_damp/reference_rate)^(reference_rate * p_step).

Without this correcion, for instance, the remaining speed difference after 0.5 seconds for 60 tps physics vs 200 tps physics is 1:1.45 (+45%))

As for the documentation, before this fix is in, this very crass diffrence in behaviours from something as inoccuous as changing the physics step time (should result in higher precision, not lower precision!) must be mentioned.

**URL to the documentation page:**
https://docs.godotengine.org/en/stable/classes/class_rigidbody3d.html#class-rigidbody3d-property-linear-damp
https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#class-projectsettings-property-physics-3d-default-linear-damp
https://docs.godotengine.org/en/stable/classes/class_rigidbody2d.html#class-rigidbody2d-property-linear-damp
https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#class-projectsettings-property-physics-2d-default-linear-damp

> If you know how to fix the issue you are reporting please consider opening a pull request.

I offer to do this later today (the documentation fix).

Further notes:
- negative values are dangerous (should get an engine fix) as the resulting damping factors are never clamped against [0..1], only against 0.
- the damping is combined in different ways for 2D and 3D. Notably, this is a difference between the 3D and 2D physics where the default values are 0.0 and 0.1 respectively, leading to a default (combined) damping of 0.1 for 3D. and 0.2 for 2D

### PR DESCRIPTION
**Pull request by thygrrr**
Correct location for the documentation changes I erroneously submitted to sphinx.
Resolves https://github.com/godotengine/godot-docs/issues/8163
Supersedes https://github.com/godotengine/godot-docs/pull/8176

Physics Damping (2D and 3D) documentation was incorrect and misleading, suggesting a critical spring damping scheme (that "tries to stop a body in a given time", but it's a "negative exponential damper", that is both velocity and tick-rate dependent).

Hopefully this will save some hairs from going grey and provide a deeper understanding for the users of this obscure value and its inner workings (Godot is a bit special in this regard, though the combine feature is really cool) :)

### CODE REVIEW

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -2072,8 +2074,11 @@
 			[/codeblocks]
 		</member>
 		<member name="physics/2d/default_linear_damp" type="float" setter="" getter="" default="0.1">
-			The default linear damp in 2D.
-			[b]Note:[/b] Good values are in the range [code]0[/code] to [code]1[/code]. At value [code]0[/code] objects will keep moving with the same velocity. Values greater than [code]1[/code] will aim to reduce the velocity to [code]0[/code] in less than a second e.g. a value of [code]2[/code] will aim to reduce the velocity to [code]0[/code] in half a second. A value equal to or greater than the physics frame rate ([member ProjectSettings.physics/common/physics_ticks_per_second], [code]60[/code] by default) will bring the object to a stop in one iteration.
+			The default linear motion damping in 3D. Damping is used to gradually slow down physical objects over time. RigidBodies will fall back to this value when combining their own damping values and no area damping value is present.
```
Comment (Talon1024) [inline doc/classes/ProjectSettings.xml:2077]: This says it's for 3D, but the original documentation says it's for 2D.

File: doc/classes/ProjectSettings.xml [hunk 2]
```diff
@@ -2072,8 +2074,11 @@
...
+			[b]Note:[/b] Godot damping calculations are velocity-dependent, meaning bodies moving faster will take a longer time to come to rest. They do not simulate inertia, friction, or air resistance. Therefore heavier or larger bodies will lose speed at the same proportional rate as lighter or smaller bodies.
+			During each physics tick, Godot will multiply the linear velocity of RigidBodies by [code]1.0 - combined_damp / physics_ticks_per_second[/code], where [code]combined_damp[/code] is the sum of the linear damp of the body and this value, or the area's value the body is in, assuming the body defaults to combine damp values. See [enum RigidBody3D.DampMode].
+			[b]Warning:[/b] Godot's damping calculations are simulation tick rate dependent. Changing [member physics/common/physics_ticks_per_second] may significantly change the outcomes and feel of your simulation. This is true for the entire range of damping values greater than 0. To get back to a similar feel, you also need to change your damp values. This needed change is not proportional and differs from case to case.
```
Comment (AThousandShips) [inline doc/classes/ProjectSettings.xml:2081]:
```suggestion
			The default linear motion damping in 2D. Damping is used to gradually slow down physical objects over time. RigidBodies will fall back to this value when combining their own damping values and no area damping value is present.
			Suggested values are in the range [code]0[/code] to [code]30[/code]. At value [code]0[/code] objects will keep moving with the same velocity. Greater values will stop the object faster. A value equal to or greater than the physics tick rate ([member physics/common/physics_ticks_per_second]) will bring the object to a stop in one iteration.		
			[b]Note:[/b] Godot damping calculations are velocity-dependent, meaning bodies moving faster will take a longer time to come to rest. They do not simulate inertia, friction, or air resistance. Therefore heavier or larger bodies will lose speed at the same proportional rate as lighter or smaller bodies.
			During each physics tick, Godot will multiply the linear velocity of RigidBodies by [code]1.0 - combined_damp / physics_ticks_per_second[/code], where [code]combined_damp[/code] is the sum of the linear damp of the body and this value, or the area's value the body is in, assuming the body defaults to combine damp values. See [enum RigidBody2D.DampMode].
			[b]Warning:[/b] Godot's damping calculations are simulation tick rate dependent. Changing [member physics/common/physics_ticks_per_second] may significantly change the outcomes and feel of your simulation. This is true for the entire range of damping values greater than 0. To get back to a similar feel, you also need to change your damp values. This needed change is not proportional and differs from case to case.
```

File: doc/classes/ProjectSettings.xml [hunk 3]
```diff
@@ -2040,9 +2040,11 @@
...
+			[b]Note:[/b] Godot damping calculations are velocity-dependent, meaning bodies moving faster will take a longer time to come to rest. They do not simulate inertia, friction, or air resistance. Therefore heavier or larger bodies will lose speed at the same proportional rate as lighter or smaller bodies.
+			During each physics tick, Godot will multiply the linear velocity of RigidBodies by [code]1.0 - combined_damp / physics_ticks_per_second[/code]. By default, bodies combine damp factors: [code]combined_damp[/code] is the sum of the damp value of the body and this value or the area's value the body is in. See [enum RigidBody2D.DampMode].
+			[b]Warning:[/b] Godot's damping calculations are simulation tick rate dependent. Changing [member physics/common/physics_ticks_per_second] may significantly change the outcomes and feel of your simulation. This is true for the entire range of damping values greater than 0. To get back to a similar feel, you also need to change your damp values. This needed change is not proportional and differs from case to case.		</member>
```
Comment (AThousandShips) [inline doc/classes/ProjectSettings.xml:2047]:
```suggestion
			[b]Warning:[/b] Godot's damping calculations are simulation tick rate dependent. Changing [member physics/common/physics_ticks_per_second] may significantly change the outcomes and feel of your simulation. This is true for the entire range of damping values greater than 0. To get back to a similar feel, you also need to change your damp values. This needed change is not proportional and differs from case to case.
		</member>
```

**Comment by thygrrr** [comment]
I don't understand the concrete failure in 
GHA / 🐧 Linux / Editor w/ Mono (target=editor) (pull_request)

**Comment by AThousandShips** [comment]
Follow my corrections above, you have a format error

**Review by Mickeon — APPROVED** [review]
This looks like a great exahustive explanation, the opposite of what it was before. Would be nice to have merged sooner than later.

**Comment by Mickeon** [comment]
If you can squash those commits it would be very nice. See https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#the-interactive-rebase

**Comment by akien-mga** [comment]
I squashed the commits, and fixed trailing spaces and the use of Windows return carriage at end of lines.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**